src/store/reducers/departmentReducer.ts

```diff
@@ -10,21 +11,21 @@ const initState: DepartmentInitState = {
 
 const departmentReducer = (state: DepartmentInitState = initState, action: ActionType) => {
   switch(action.type) {
-    case departmentActionTypes.FETCH_DEPARTMENT_BY_ID:
+    case APIActionTypes.FETCH_START:
       return {
         ...state,
         loading: true
       };
 
-    case departmentActionTypes.FETCH_DEPARTMENT_BY_ID_SUCCESS:
+    case departmentActionTypes.FETCH_DEPARTMENT_BY_ID:
       return {
         ...state,
         loading: false,
         department: action.payload,
         errorMessage: ''
       };
 
-    case departmentActionTypes.FETCH_DEPARTMENT_BY_ID_FAILURE:
+    case APIActionTypes.FETCH_FAILURE:
```

Comment:
no need to put such cases into each reducer. Create a separated "loadingReducer" and it will be enough. Otherwise you will have multiple state mutations